### ISSUE TITLE
Forced Closing

### ISSUE THREAD

**Issue by Cruzan80** [weto91/kindle-cover-fixer#16]
When I load the program, it force closes almost immediately. It does not matter if the Kindle is plugged in or not. Kindle Scribe, Windows 10.

**Comment by weto91** [comment]
Hello, have you installed Calibre with any Calibre library and books in it?

**Comment by Cruzan80** [comment]
Yes, Calibre was open at the time. Several thousand books in it. Located on
a NAS (mapped network drive).

**Comment by weto91** [comment]
Okey, maybe thats the problem. The application does not manage network mappings... So you must transfer the library to one of the local partitions to run this application...

Regards.

**Comment by Cruzan80** [comment]
I will try that. Currently, I have it mapped to show up as a standard drive
letter, so it "should" see it the same as another physical drive. Will the
app handle Libraries other than in C:?

**Comment by weto91** [comment]
No, if is mapped with drive letter, thats not the problem. Can you attach :%APPDATA%/Calibre/gui.json file? in this file, are listed all Calibre libraries, That way I see how it is exposed and we start looking for the problem from there.

Thanks.

**Comment by Cruzan80** [comment]
Sorry for the delay. I was incorrect, it wasn't mapped to a drive for that
instance. I made a "test" library on an internal D: drive, and transferred
some files to it. The program was able to work eventually, but kept
crashing at random times. Using the clear tools (both Calibre and Kindle)
seemed to help, as well as not having Calibre open helped as well.
Sometimes it would crash when scanning the titles, sometimes when
generating covers. Is there a practical field limit? The "test" library had
a few hundred files. No real specific instance of when it would crash,
sometimes immediately, sometimes in the middle of a scan or generating
covers.

**Comment by weto91** [comment]
If you have titles with strange symbols, it is possible that the application could crash. During development, tests were carried out with a wide variety of titles, but it is possible that there is a title that contains a symbol that we have not handled and fails.

In principle, it doesn't matter how long the title is, if caliber allows a length, so does this application.

If it fails again, send me the application log file and we can try to figure out what's happening (%APPDATA%/Kindle Cover Fixer/*.log)

Thank you.

**Comment by SlowWap** [comment]
Fixed by adding a 'connection.open' test in the Libraries.cs file. Test is valid database can be opened (mapped drive, network URL, etc...). Quick and dirty code so expection is not thrown in ImageCover.cs...

//test if library can be opened
try
{
    //
    string data = string.Empty;
    string cs = @"URI=file:" + library + @"\metadata.db";
    using SQLiteConnection connection = new(cs);
    connection.Open();
    //
    connection.Close();

    //Add library to list
    if (!library.Contains("[ERROR]"))
    {
        libraryPath.Items.Add(library);
    }
    else
    {
        DisableControl(findBooks);
        DisableControl(generateButton);
        MessageBox.Show(Strings.CheckLibrary, Strings.CheckLibraryTitle, MessageBoxButton.OK, MessageBoxImage.Error);
    }
    LogLine("LIBRARY", "Library: " + library + " loaded.");

}
catch (Exception)
{
    //don't add lib to list
    ;
}

**Comment by SlowWap** [comment]
Also need a busy wait on... CheckKindleType() timed event, so that is does not fire when in FindBookTasks or TransferFilesToKindle because MTP devices will throw an exception if busy doing file transfer while an enumeration is attempted.
